src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/LinkDesignerBase.cs

```diff
@@ -198,19 +201,55 @@ protected void FixupAssemblyTypes (AssemblyDefinition assembly, TypeDefinition d
 
 		protected override void ProcessAssembly (AssemblyDefinition assembly)
 		{
+			allAssemblies.Add (assembly);
 			LoadDesigner ();
 
 			var action = Annotations.HasAction (assembly) ? Annotations.GetAction (assembly) : AssemblyAction.Skip;
 			if (action == AssemblyAction.Delete)
 				return;
 
 			if (ProcessAssemblyDesigner (assembly)) {
-				if (action == AssemblyAction.Skip || action == AssemblyAction.Copy)
+				if (action == AssemblyAction.Skip || action == AssemblyAction.Copy) {
 					Annotations.SetAction (assembly, AssemblyAction.Save);
+					processedAssemblies.Add (assembly);
+				}
+			}
+		}
+
+		internal void ProcessNonDesignerAssemblies ()
+		{
+			
+		}
+
+		protected override void EndProcess ()
+		{
+			if (processedAssemblies.Count > 0) {
+				foreach (var assembly in allAssemblies) {
+					if (processedAssemblies.Contains (assembly))
+						continue;
+
+					var action = Annotations.HasAction (assembly) ? Annotations.GetAction (assembly) : AssemblyAction.Skip;
+					if (action == AssemblyAction.Delete)
+						continue;
+
+					foreach (var processedAssembly in processedAssemblies) {
+						if (assembly.MainModule.AssemblyReferences.Any (r => r.FullName == processedAssembly.Name.FullName)) {
```

Comment:
I wish there was a more efficient way of doing this. But I can't think of anything right now :( I guess fixing the issue is more important at this point.